velox/dwio/dwrf/reader/ColumnReader.cpp

```diff
@@ -1720,17 +1856,34 @@ ListColumnReader::ListColumnReader(
       requestedType_{requestedType} {
   DWIO_ENSURE_EQ(nodeType_->id, dataType->id, "working on the same node");
   EncodingKey encodingKey{nodeType_->id, flatMapContext_.sequence};
-  // count the number of selected sub-columns
-  RleVersion vers = convertRleVersion(stripe.getEncoding(encodingKey).kind());
 
-  auto lenId = encodingKey.forKind(proto::Stream_Kind_LENGTH);
-  bool vints = stripe.getUseVInts(lenId);
-  length = createRleDecoder</*isSigned*/ false>(
-      stripe.getStream(lenId, true),
-      vers,
-      memoryPool_,
-      vints,
-      dwio::common::INT_BYTE_SIZE);
+  if (stripe.format() == DwrfFormat::kDwrf) {
+    // count the number of selected sub-columns
+    RleVersion vers = convertRleVersion(stripe.getEncoding(encodingKey).kind());
+
+    auto lenId = encodingKey.forKind(proto::Stream_Kind_LENGTH);
```

Comment:
Are vers and lenId the only difference for orc and dwrf? If so, can we simply this if-else clause?